src/mem/ruby/protocol/GPU_VIPER-TCC.sm

```diff
@@ -815,6 +827,28 @@ machine(MachineType:TCC, "TCC Cache")
     }
   }
 
+  action(pd_sendProbeResponseDowngrade, "pd", desc="send probe downgrade") {
+    enqueue(responseToNB_out, ResponseMsg, 1) {
+      out_msg.addr := address;
+      out_msg.Type := CoherenceResponseType:CPUPrbResp;  // TCC, L3  respond in same way to probes
+      out_msg.Sender := machineID;
+      out_msg.Destination.add(mapAddressToMachine(address, MachineType:Directory));
+      if (getState(tbe, cache_entry, address) == State:V || getState(tbe, cache_entry, address) == State:M || getState(tbe, cache_entry, address) == State:W) {
```

Comment:
this is not my favorite format, but I guess in this case checking isDirty or something like that is not sufficient?  If so, then this is fine.

Reply:
The TBE/cache entry's dirty flag doesn't seem to be set anywhere. I would have to set that to true/false in transitions before I can use it here. Would you prefer I do that, or is this sufficient?

Reply:
let it be then